lambda/function.go

```diff
@@ -16,13 +16,31 @@ import (
 // Function struct which wrap the Handler
 type Function struct {
 	handler Handler
+	context context.Context
 }
 
 // NewFunction which creates a Function with a given Handler
 func NewFunction(handler Handler) *Function {
 	return &Function{handler: handler}
 }
 
+// NewFunctionWithContext which creates a Function with a given Handler and sets the base Context.
+func NewFunctionWithContext(ctx context.Context, handler Handler) *Function {
+	return &Function{
+		context: ctx,
+		handler: handler,
+	}
+}
+
+// Context returns the base context used for the fn.
+func (fn *Function) Context() context.Context {
```

Comment:
This should be private, or moved inside of `Invoke`. The public visibility of the other `Function` methods is done only as a requirement of the `net/ipc` library.

Reply:
@bmoffatt made this un exported :+1: